tests/test_mechanical.py

```diff
@@ -510,6 +513,93 @@ def test_launch_grpc_not_supported_version():
         pymechanical.mechanical.launch_grpc(exec_file=exec_file)
 
 
+@pytest.mark.remote_session_launch
+def test_warning_message_pythonnet():
+    """Test pythonnet warning of the remote session in virtual env."""
+    venv_name = "pythonnetvenv"
+    base = os.getcwd()
+
+    if "win" in sys.platform:
+        exe_dir = "Scripts"
+    else:
+        exe_dir = "bin"
+
+    venv_bin = os.path.join(base, "." + venv_name, exe_dir)
+
+    # Set up path to use the virtual environment
+    original_path = os.environ["PATH"]
+    os.environ["PATH"] = venv_bin + os.pathsep + os.environ.get("PATH", "")
+
+    # Create virtual environment
+    subprocess.run([sys.executable, "-m", "venv", "." + venv_name])
+
+    # Upgrade pip
+    print("Upgrading pip")
+    upgrade_pip = subprocess.Popen(
+        [os.path.join(venv_bin, "python"), "-m", "pip", "install", "-U", "pip"]
+    )
+    upgrade_pip.wait()
+
+    # Install tests
+    print("Installing tests")
+    install_tests = subprocess.Popen([os.path.join(venv_bin, "pip"), "install", "-e", ".[tests]"])
+    install_tests.wait()
+
+    # Install pythonnet
+    print("Installing pythonnet")
+    install_pythonnet = subprocess.Popen([os.path.join(venv_bin, "pip"), "install", "pythonnet"])
+    install_pythonnet.wait()
+
+    # Run remote session in virtual env with pythonnet installed
+    print("Running the remote session in a virtual environment")
+    remote_py = os.path.join(base, "tests", "scripts", "run_remote_session.py")
+    check_warning = subprocess.Popen(
+        [os.path.join(venv_bin, "python"), remote_py], stderr=subprocess.PIPE
+    )
+    check_warning.wait()
+    stderr_output = check_warning.stderr.read().decode()
+
+    """If UserWarning & pythonnet are in the stderr output,
+    set warning to 0. Otherwise, set warning to 1"""
```

Comment:
Better use in-line comments
```suggestion
   # If UserWarning & pythonnet are in the stderr output, set warning to True.
   # Otherwise, set warning to False
```